src/api/Nhs.Appointments.Core/Bookings/BookingAvailabilityStateService.cs

```diff
@@ -178,94 +259,14 @@ private BookingAvailabilityState BuildState(IEnumerable<Booking> bookings, List<
         return state;
     }
 
-    private async Task<BookingAvailabilityState> BuildState(
-        string site, 
-        DateTime from, 
-        DateTime to,
-        BookingAvailabilityStateReturnType returnType, 
-        Session matcher = null, 
-        Session replacement = null,
-        bool? isWildCard = null)
+    private static AvailabilityUpdateProposalAction DetermineAvailabilityUpdateProposalAction(bool hasMatcherAndReplacementSessions)
```

Comment:
Again this method wouldn't be necessary with an `AvailabilityChange` class as it could be a computed property